invokeai/app/services/config.py

```diff
@@ -0,0 +1,290 @@
+    esrgan              : bool = Field(default=True, description="Enable/disable upscaling code")
+    #fmt: on
+
+    def __init__(self, conf: DictConfig = None, argv: List[str]=None, **kwargs):
+        '''
+        Initialize InvokeAIAppconfig.
+        :param conf: alternate Omegaconf dictionary object
+        :param argv: aternate sys.argv list
+        :param **kwargs: attributes to initialize with
+        '''
+        super().__init__()
+        
+        # Set the runtime root directory. We parse command-line switches here
+        # in order to pick up the --root_dir option.
+        self.parse_args(argv)
+        if not self.root:
+            self.root = self._find_root()
+        if not conf:
+            try:
+                conf = OmegaConf.load(self.root_dir / INIT_FILE)
+            except:
+                pass
+        InvokeAISettings.initconf = conf
+        # parse args again in order to pick up settings in configuration file
+        self.parse_args(argv)
+
+        # restore initialization values
+        for k in kwargs:
+            setattr(self,k,kwargs[k])
+
+    @property
+    def root_dir(self)->Path:
+        return self.root.expanduser()
+
+    def _resolve(self,partial_path:Path)->Path:
+        return (self.root_dir / partial_path).resolve()
+
+    @property
+    def output_path(self)->Path:
+        return self._resolve(self.outdir)
+
+    @property
+    def model_conf_path(self)->Path:
+        return self._resolve(self.conf)
+
+    @property
+    def embedding_path(self)->Path:
+        return self._resolve(self.embedding_dir) if self.embedding_dir else None
+
+    @property
+    def autoconvert_path(self)->Path:
+        return self._resolve(self.autoconvert_dir) if self.autoconvert_dir else None
+
+    @property
+    def gfpgan_model_path(self)->Path:
+        return self._resolve(self.gfpgan_model_dir) if self.gfpgan_model_dir else None
+
+    @staticmethod
+    def find_root()->Path:
+        if os.environ.get("INVOKEAI_ROOT"):
+            root = Path(os.environ.get("INVOKEAI_ROOT")).resolve()
+        elif (
+                os.environ.get("VIRTUAL_ENV")
+                and Path(os.environ.get("VIRTUAL_ENV"), "..", INIT_FILE).exists()
+        ):
+            root = Path(os.environ.get("VIRTUAL_ENV"), "..").resolve()
+        else:
+            root = Path("~/invokeai").expanduser().resolve()
+        return root
+
+
+class InvokeAIWebConfig(InvokeAIAppConfig):
+    '''
+    Web-specific settings
+    '''
+    #fmt: off
+    type               : Literal["web"] = "web"
+    allow_origins      : List = Field(default=[], description='Allowed CORS origins')
+    allow_credentials  : bool = Field(default=True, description='Allow CORS credentials')
+    allow_methods      : List = Field(default=["*"], description='Methods allowed for CORS')
+    allow_headers      : List = Field(default=["*"], description='Headers allowed for CORS')
```

Comment:
`invokeai-node-web --host 0.0.0.0` throws `unrecognized argument: --host`, etc

```suggestion
    allow_origins      : List = Field(default=[], description="Allowed CORS origins")
    allow_credentials  : bool = Field(default=True, description="Allow CORS credentials")
    allow_methods      : List = Field(default=["*"], description="Methods allowed for CORS")
    allow_headers      : List = Field(default=["*"], description="Headers allowed for CORS")
    host               : str = Field(default="127.0.0.1", description="IP address to bind to")
    port               : int = Field(default=9090, description="Port to bind to")
```

Reply:
(only 290-291 are germane to the issue - other lines just change single quotes to doubles for consistency)

Reply:
Thanks!